File: bench/fixtures/01-python-typeerror/bug.py
PROMOS = {"SAVE10": 0.10, "SAVE20": 0.20}


def apply_discount(promo_code, price):
    return price * (1 - PROMOS[promo_code])


def line_total(item):
    base = item["price"] * item["qty"]
    return apply_discount(item["promo"], base)


def cart_total(items):
    return sum(line_total(i) for i in items)


if __name__ == "__main__":
    cart = [
        {"price": 10, "qty": 2, "promo": "SAVE10"},
        {"price": 5, "qty": 1},
    ]
    print(cart_total(cart))

File: bench/fixtures/06-ts-undefined-call/bug.ts
type Item = { price: number; qty: number; promo?: { code: string; pct: number } };

function applyPromo(promo: { pct: number }, total: number): number {
  return total * (1 - promo.pct);
}

function lineTotal(item: Item): number {
  const base = item.price * item.qty;
  return applyPromo(item.promo!, base);
}

const cart: Item[] = [
  { price: 10, qty: 2, promo: { code: "SAVE10", pct: 0.1 } },
  { price: 5, qty: 1 },
];

console.log(cart.reduce((sum, i) => sum + lineTotal(i), 0));
